### PR TITLE
Multiarch Tuning Operator tests on different OCP versions

### PR DESCRIPTION
**Pull request by aleskandro**
This is just a POC to continue to foster a discussion on how we should handle testing of the Multiarch Tuning operator across different versions of OCP. 

- We must keep an eye on both the OLM and Kustomize-based deployments as it happens we miss some issues in one of the cases that affect the other and finally, the deployment from a catalog: just recently, we've faced a problem with the operator-sdk applying privileged pod security policies when using `operator-sdk run bundle --security-context-config restricted ...`. The bundle-based deployment escaped the enforcement of checks for mounting hostpath volumes. The issue was caught by the subsequent runs of the kustomize-based deployments (on 4.17).
- This is a platform-aligned deployment, so we must support it on all the supported OCP versions (starting 4.16). Therefore, we'd ensure the operator gets tested on them.
- Tests involving node configuration changes are inherently slow as the Machine Config Operator drains or reboots nodes to apply that configuration. Beyond any further optimizations to reduce the time-to-test, I'm considering a tag in the slow tests that we can use to skip their execution. I'm unsure of the result, but I'm proposing an approach consisting of running the test suite twice in the same cluster: the first time with the bundle-based deployment, and the second time with the kustomize-based one. To avoid executing the slower tests twice per run, I'm letting them run only in one of the two cases. We randomly choose whether to run them on the first or second run to have a signal from both.
- I'm changing the names of the jobs to be shorter
- I'm setting the e2e + origin test run as the default post-submit, and doing one post-submit per version.

Some of the changes and optimization will be effective after the work in https://github.com/openshift/multiarch-tuning-operator/pull/193, including the patch at https://github.com/openshift/multiarch-tuning-operator/pull/193#issuecomment-2316266200

Ideally, we should use cluster_claims instead of completely provisioning clusters to reduce time and issues due to installation, but this shouldn't change the strategy and jobs definition (after installation).

/cc @lwan-wanglin @AnnaZivkovic @Prashanth684

### CODE REVIEW

**Comment by Prashanth684** [comment]
is it possible to change the names of the jobs to accurately reflect which ones use OLM and which ones use kustomize?

**Comment by Prashanth684** [comment]
> We must keep an eye on both the OLM and Kustomize-based deployments as it happens we miss some issues in one of the cases that affect the other and finally

Can we make it so OLM based deploys are tested for all releases and kustomize is tested for the current release? since we have only one release stream that should still catch issues when deploying without OLM?

**Comment by Prashanth684** [comment]
why did we remove the periodic? do we not want one for our operator?

File: ci-operator/config/openshift/multiarch-tuning-operator/openshift-multiarch-tuning-operator-main.yaml [hunk 1]
```diff
@@ -223,53 +245,73 @@ tests:
...
-- as: e2e-aws-post-multi-operator-olm
+    workflow: openshift-e2e-aws-ovn
+- as: e2e-aws-ovn-mto-origin-4-17-post
   cluster: build10
   postsubmit: true
-  skip_if_only_changed: (^docs/)|((^|/)OWNERS(_ALIASES)?$)|((^|/)[A-Za-z]+\.md$)|((^|/)\.github/)|((^|/)\.tekton/)|((^|/).*konflux\.Dockerfile$)
+  skip_if_only_changed: (^docs/)|((^|/)OWNERS(_ALIASES)?$)|((^|/)[A-Za-z]+\.md$)|((^|/)\.github/)|((^|/)\.tekton/)|((^|/)konflux\.Dockerfile$)
   steps:
-    cluster_profile: aws
+    cluster_profile: gcp
```
Comment (lwan-wanglin) [inline ci-operator/config/openshift/multiarch-tuning-operator/openshift-multiarch-tuning-operator-main.yaml:254]:
```suggestion
    cluster_profile: aws
```

File: ci-operator/config/openshift/multiarch-tuning-operator/openshift-multiarch-tuning-operator-main.yaml [hunk 2]
```diff
@@ -179,32 +192,41 @@ tests:
...
+    test:
+    - ref: ipi-install-heterogeneous
+    - ref: multiarch-tuning-operator-e2e-olm-kustomize
+    workflow: ipi-gcp
+- as: e2e-aws-ovn-mto-origin-4-16-post
+  cluster: build10
+  postsubmit: true
+  skip_if_only_changed: (^docs/)|((^|/)OWNERS(_ALIASES)?$)|((^|/)[A-Za-z]+\.md$)|((^|/)\.github/)|((^|/)\.tekton/)|((^|/)konflux\.Dockerfile$)
+  steps:
+    cluster_profile: gcp
```
Comment (lwan-wanglin) [inline ci-operator/config/openshift/multiarch-tuning-operator/openshift-multiarch-tuning-operator-main.yaml:221]:
```suggestion
    cluster_profile: aws
```

File: ci-operator/step-registry/multiarch/tuning-operator/e2e/olm-kustomize/multiarch-tuning-operator-e2e-olm-kustomize-commands.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# execute the slow tests only on one of the runs (choose randomly)
+if [ $((RANDOM % 100)) -lt 50 ]; then
+  export GINKGO_EXTRA_ARGS="--skip=\\[slow-30\\]"
+  echo "[SKIP] Skipping slow tests in the OLM deployment"
+fi
+USE_OLM=true ./hack/deploy-and-e2e.sh
+if [ "${GINKGO_EXTRA_ARGS}" == "" ]; then
+  export GINKGO_EXTRA_ARGS="--skip=\\[slow-30\\]"
+  echo "[SKIP] Skipping slow tests in the Kustomize deployment"
+fi
```
Comment (lwan-wanglin) [inline ci-operator/step-registry/multiarch/tuning-operator/e2e/olm-kustomize/multiarch-tuning-operator-e2e-olm-kustomize-commands.sh:12]: whether we should also reset `GINKGO_EXTRA_ARGS=""` if it has been set in olm case?

Comment (lwan-wanglin) [inline ci-operator/step-registry/multiarch/tuning-operator/e2e/olm-kustomize/multiarch-tuning-operator-e2e-olm-kustomize-commands.sh:12]: whether we should reset `GINKGO_EXTRA_ARGS=""` if if was set in olm testing?

Comment (aleskandro) [inline ci-operator/step-registry/multiarch/tuning-operator/e2e/olm-kustomize/multiarch-tuning-operator-e2e-olm-kustomize-commands.sh:12]:
```suggestion
else
  GINKGO_EXTRA_ARGS=""
fi
```

Comment (aleskandro) [reply ci-operator/step-registry/multiarch/tuning-operator/e2e/olm-kustomize/multiarch-tuning-operator-e2e-olm-kustomize-commands.sh:14]: yes. done.

**Comment by aleskandro** [comment]
@lwan-wanglin fixed your comments. I would like more to have an opinion about the 'strategy' though, especially since the actual cluster provisioning will change with cluster_claims. Are we all ok with the points in the description of the PR?

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-17

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-17

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-16

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-17

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-16

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-18

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-16

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-16

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-18

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-e2e-gcp-4-18

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-ocp417-e2e-gcp

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-ocp416-e2e-aws-ovn-mto-origin-post-verify

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-ocp418-e2e-aws-ovn-mto-origin-post-verify

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-ocp417-e2e-aws-ovn-mto-origin-post-verify

**Comment by lwan-wanglin** [comment]
/pj-rehearse pull-ci-openshift-multiarch-tuning-operator-main-ocp417-e2e-aws-ovn-mto-origin-post-verify

**Pull request merged**